Bug: E5012 error shows 'RETURN_VALUE' instead of actual return type

## Version

v0.24.0

## Bug Description

When a function returns multiple values (tuple) and the caller tries to assign it to a single variable with a mismatched type, the error message shows `RETURN_VALUE` as the actual type instead of displaying the real return type.

## Steps to Reproduce

1. Create a function that returns a tuple, e.g., `([byte], Error)`
2. Try to assign the result to a single variable with type `[byte]`

```ez
// In library file
do get_file_data(f string) -> ([byte], Error){
    temp data [byte]
    temp err Error = error("")
    data = io.read_bytes(f)
    if err != nil {
        return data, err
    }
    return data, nil
}

// In main file
do main(){
    temp data [byte] = lib.get_file_data(DB_FILE)  // Error here
}
```

## Actual Behavior

```
error[E5012]: return type mismatch: expected [byte], got RETURN_VALUE
  --> main/main.ez:9:27
   |
9 |     temp data [byte] = lib.get_file_data(DB_FILE)
   |                           ^ error occurred here
   |
```

## Expected Behavior

The error should show the actual return type:
```
error[E5012]: return type mismatch: expected [byte], got ([byte], Error)
```

## Analysis

The `objectTypeToEZ()` function in `pkg/interpreter/evaluator.go` is likely not handling `ReturnValue` objects properly and falling back to showing the object type name (`RETURN_VALUE`) instead of resolving the actual contained type(s).

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -2688,6 +2688,17 @@ func objectTypeToEZ(obj Object) string {
 		return "nil"
 	case *Function:
 		return "function"
+	case *ReturnValue:
+		// Handle return values - format as tuple if multiple values
+		if len(v.Values) == 1 {
+			return objectTypeToEZ(v.Values[0])
+		}
+		// Multiple return values - format as tuple
+		types := make([]string, len(v.Values))
+		for i, val := range v.Values {
+			types[i] = objectTypeToEZ(val)
+		}
+		return "(" + strings.Join(types, ", ") + ")"
 	default:
 		return string(obj.Type())
 	}

